File: app/src/main/java/com/kipucode/data/local/dao/relation/CourseWithLessons.java
    package com.kipucode.data.local.dao.relation;

    import androidx.room.Embedded;
    import androidx.room.Relation;

    import com.kipucode.data.local.model.Course;
    import com.kipucode.data.local.model.Lesson;

    import java.util.List;

    public class CourseWithLessons {
        @Embedded
        public Course course;
        @Relation(
            parentColumn = "id",
            entityColumn = "course_id"
        )
        public List<Lesson> lessons;
    }

File: app/src/main/java/com/kipucode/data/local/dao/relation/CourseWithLessons.kt
package com.kipucode.data.local.dao.relation

import androidx.room.Embedded
import androidx.room.Relation
import com.kipucode.data.local.model.CourseEntity
import com.kipucode.data.local.model.LessonEntity

data class CourseWithLessons(
    @Embedded
    val course: CourseEntity,
    @Relation(
        parentColumn = "id",
        entityColumn = "course_id"
    )
    val lessons: List<LessonEntity>
)
